enumerables.rb

```diff
@@ -0,0 +1,171 @@
+def my_each
+  return enum_for(__callee__) unless block_given?
+
+  size.times { |i| yield to_a[i] }
+end
+
+def my_each_with_index
+  return enum_for(__callee__) unless block_given?
+
+  length.times { |i| yield(to_a[i], i) }
+end
+
+def my_select
+  return enum_for(__callee__) unless block_given?
+
+  res = []
+  my_each { |i| res << i if yield(i) }
+  res
+end
+
+def my_all
+  return enum_for(__callee__) unless block_given?
+
+  my_each { |i| return false unless yield(i) }
+  true
+end
+
+def my_any
+  return enum_for(__callee__) unless block_given?
+
+  my_each { |i| return true if yield(i) }
+  false
+end
+
+def my_none
+  return enum_for(__callee__) unless block_given?
+
+  my_each { |i| return false if yield(i) }
+  true
+end
+
+def my_count
+  return enum_for(__callee__) unless block_given?
+
+  c = 0
+  my_each { |i| c += 1 if yield(i) }
+  c
+end
```

Comment:
### Fix the failing error
![count](https://user-images.githubusercontent.com/8114528/84962362-5bbd9b80-b10f-11ea-956c-4736af180a69.png)

Reply:
Updated function to receive parameters